packages/base-map/src/index.tsx

```diff
@@ -0,0 +1,192 @@
+  onClick?: (evt: Event) => void;
+  // Unknown is used here because of a maplibre/mapbox issue with the true type, MapLayerMouseEvent
+  onContextMenu?: (e: unknown) => void;
+  // TODO: does this cause integration issues?
+  onViewportChanged?: (e: maplibregl.MapLibreEvent) => void;
+  passedRef?: React.Ref<MapRef>;
+  zoom?: number;
+};
+type State = {
+  latitude: number;
+  longitude: number;
+  zoom: number;
+  fitBoundsOptions?: Record<string, number | string | boolean>;
+};
+
+const BaseMap = ({
+  // These tiles are free to use, but not in production
+  baseLayer = "https://basemaps.cartocdn.com/gl/positron-gl-style/style.json",
+  center,
+  children,
+  forceMaxHeight,
+  mapLibreProps,
+  maxZoom,
+  onClick,
+  onContextMenu,
+  passedRef,
+  onViewportChanged,
+  zoom: initZoom = 12
+}: Props): JSX.Element => {
+  const [viewState, setViewState] = React.useState<State>({
+    fitBoundsOptions: {
+      animate: true,
+      duration: 300,
+      essential: false,
+      maxDuration: 600,
+      padding: 200
+    },
+    latitude: center?.[0],
+    longitude: center?.[1],
+    zoom: initZoom
+  });
+  const [fakeMobileHover, setFakeMobileHover] = useState(false);
+
+  useEffect(() => {
+    callIfValid(onViewportChanged)(viewState);
+  }, [viewState]);
+
+  useEffect(() => {
+    if (center?.[0] === null || center?.[1] === null) return;
+
+    setViewState({
+      ...viewState,
+      latitude: center?.[0],
+      longitude: center?.[1]
+    });
+  }, [center]);
+
+  const toggleableLayers = Array.isArray(children)
+    ? children
+        .flat()
+        .filter(child => child?.props?.id !== undefined)
+        .map(child => {
+          const { visible, name, id } = child.props;
+          return { visible, name, id };
+        })
+    : [];
+
+  const [hiddenLayers, setHiddenLayers] = useState(
+    toggleableLayers.filter(layer => !layer?.visible).map(layer => layer.id)
+  );
+
+  const adjustHiddenLayers = useCallback(
+    id => {
+      const updatedLayers = [...hiddenLayers];
+      // Delete the layer id if present, add it otherwise
+      updatedLayers.includes(id)
+        ? updatedLayers.splice(updatedLayers.indexOf(id), 1)
+        : updatedLayers.push(id);
+
+      setHiddenLayers(updatedLayers);
+    },
+    [hiddenLayers]
+  );
+
+  return (
+    <MapProvider>
+      <Map
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...mapLibreProps}
+        ref={passedRef}
+        id="mainMap"
+        latitude={viewState.latitude}
+        longitude={viewState.longitude}
+        mapLib={maplibregl}
+        mapStyle={baseLayer}
+        maxZoom={maxZoom}
+        onClick={onClick}
+        onContextMenu={onContextMenu}
+        onMove={evt => setViewState(evt.viewState)}
+        onTouchStart={() => {
+          setFakeMobileHover(false);
+        }}
+        style={{
+          display: "block",
+          height: forceMaxHeight ? "90vh" : "100%",
+          width: "100%"
+        }}
```

Comment:
I wanted to avoid repeating the same inline style in the stories, but story-based styled component might work... 

`100%` doesn't work in the storybook unfortunately.